Suma proxy upgrade 3.2 to 4 #7843 ke (#473)
SUSE/doc-susemanager: modules/upgrade/pages/proxy-migration.adoc

[[proxy-migration]]
= Proxy Migration

// also see client-migration.adoc

> Upgrading from {susemgrproxy} 3.2 with the latest service pack (SP) to version 4 can be automated, but requires some preparation steps.

Reviewer: ```suggestion
Upgrading from {productname} Proxy 3.2 with the latest service pack (SP) to version 4 can be automated, but requires some preparation steps.
```

////
To upgrade the SP version on SLE{nbsp}12 (for example, upgrading from SLE{nbsp}12 or any SLE{nbsp}12{nbsp}SPx to SLE{nbsp}12{nbsp}SP4) can be fully automated and requires no additional preparation.
////

== Preliminary Considerations

The recommended order for migrations is to first migrate the {productname} server and then the proxies.

> .Using SUSE Manager proxy 3.2 with SUSE Manager 4

Reviewer: ```suggestion
.Using {productname} Proxy 3.2 with {productname} 4
```

Author: Yes, I was a little bit lazy yesterday ;-(

[NOTE]
====

> A SUSE Manager proxy 3.2 cooperates just fine with a new SUSE Manager 4. After the server has been migrated to version 4, proxies can be autoupgraded from the SUSE Manager server with YaST autoinstallation.

Reviewer: ```suggestion
{productname} Proxy 3.2 is compatible with {productname} Server 4. 
After the server has been migrated to version 4, proxies can be automatically upgraded from the SUSE Manager server using YaST autoinstallation.
```

====

> Right now, migrating SUSE Manager proxies 3.2 only works for proxies that have been registered as traditional clients.

Reviewer: ```suggestion
Migrating {productname} Proxy 3.2 only works for proxies that have been registered as traditional clients.
```

Proxies that are Salt minions cannot be migrated to the new version this way.

[[proxy-migration-inplace]]
== Upgrading SUSE Manager Proxy In-place

[[proxy-migration-inplace-prep]]

> === Preparation

Reviewer: I would remove the heading here (to avoid the double heading), the Procedure can stand on its own.

Author: agreed, makes sense.

> .Procedure: Preparation

Reviewer: ```suggestion
.Procedure: Preparing to Upgrade
```

Procedures should ideally have a gerund title

> . First you need to create an autoinstallable distribution based on SLE 15 SP1 because the SUSE Manager Proxy 4 product is based on this version.

Reviewer: ```suggestion
. Create an autoinstallation distribution based on SLE 15 SP1.
This is required because {productname} Proxy 4 is based on this version.
```

> For creating an autoinstallable distribution, see xref:#ref.webui.systems.autoinst.distribution (menu:Main Navigation[Systems > Autoinstallation > Distributions]).

Reviewer: ```suggestion
For information on creating an autoinstallation distribution, see xref:#ref.webui.systems.autoinst.distribution (menu:Main Navigation[Systems > Autoinstallation > Distributions]).
```

Author: @Loquacity agreed, but in the webui we'd better fix the string accordingly.  Here, for example:

https://suma-refhead-srv.mgr.suse.de/rhn/kickstart/ViewTrees.do

(ATM, I'm not able to take a screen shot...  Leap 15.1 betaX installation...)

. Create an autoinstallation profile.
For more information, see xref:#proxy-migration-inplace-profile.

> In the following template example the label [literal]``proxy4`` is used for both the autoinstallable distribution and for the autoinstall profile.

Reviewer: ```suggestion
This section includes a template autoinstallation profile for you use.
In the example, the label [literal]``proxy4`` is used for both the autoinstallation distribution and for the autoinstallation profile.
```

Author: I commited a slightly different patch.

    <modified config:type="boolean">true</modified>
    <remove_old config:type="boolean">false</remove_old>
  </backup>
  <networking>
    <keep_install_network config:type="boolean">true</keep_install_network>
    <start_immediately config:type="boolean">true</start_immediately>
  </networking>
  <software>
    <image/>
    <install_recommended config:type="boolean">true</install_recommended>
    <instsource/>
    <patterns config:type="list">
      <pattern>suma_proxy</pattern>
    </patterns>
    <products config:type="list">
      <listentry>SUSE-Manager-Proxy</listentry>
    </products>
  </software>
  <scripts>
    <chroot-scripts config:type="list">
      <script>
        <filename>cleanup.sh</filename>
        <chrooted config:type="boolean">true</chrooted>
        <interpreter>shell</interpreter>
        <source>
        <![CDATA[
#!/bin/sh
         sysconf_addword /etc/sysconfig/apache2 APACHE_MODULES wsgi
         if [ -e /etc/rhn/rhn.conf.rpmsave ]; then
             cp /etc/rhn/rhn.conf.rpmsave /etc/rhn/rhn.conf
         fi
         zypper rr --all
         ]]>
        </source>
      </script>
    </chroot-scripts>
  </scripts>
</profile>
----

.Procedure

> . Make sure all the channels referenced in this autoinstallation profile are available and fully synced.

Reviewer: ```suggestion
. Make sure all the channels referenced in this autoinstallation profile are available and fully synchronized.
```

. Replace the label [literal]``proxy4`` with the one you use for your autoinstallation profile.

> === Running the In-place Upgrade

Reviewer: Make this the Procedure title.

.Procedure

> . In the {webui}, upload the edited autoinstallation profile to your {productname} server (menu:Main Navigation[Systems > Autoinstallation > Profiles]).

Reviewer: ```suggestion
. In the {productname} {webui}, upload the edited autoinstallation profile to your {productname} server (menu:Main Navigation[Systems > Autoinstallation > Profiles]).
```

> . In the [guimenu]``Kernel Options`` field, enter the following value:

Reviewer: ```suggestion
. In the [guimenu]``Kernel Options`` field, enter this value:
```

+
----
autoupgrade=1 Y2DEBUG=1
----
+

> The debug setting is not required but can help investigating problems in case something goes wrong.

Reviewer: ```suggestion
The debug setting is not required but can help with investigating future problems you might encounter.
```

> The [literal]``autoupgrade` parameter however is mandatory.

Reviewer: ```suggestion
The [literal]``autoupgrade` parameter, however, is mandatory.
```

> . Select the proxy from the system list in SUSE Manager, click the [guimenu]``Provisioning`` tab and select the profile just created.

Reviewer: ```suggestion
. Select the proxy from the system list in {productname}, click the [guimenu]``Provisioning`` tab, and select the profile you created earlier.
```

. Click btn:[Start Autoinstallation].

> The system will download the needed files, change the bootloader entries, reboot and will start the automatic upgrade.

Reviewer: ```suggestion
The system will download the needed files, change the bootloader entries, reboot, and start the upgrade.
```

=== Cleaning Up After the In-place Upgrade

After the proxy has completed the automatic upgrade process, it still will have the old channels from SUSE Manager assigned.

> So in case updates are reported for the proxy, do not apply them.

Reviewer: ```suggestion
If the proxy reports that there are updates available, do not apply them.
```

> Instead select the proxy from the system list, click change channel assignment. Then deselect the old channels and assign the new ones that have also been used for the upgrade itself.

Reviewer: ```suggestion
Select the proxy from the system list and click [guimenu]``change channel assignment``. 
Clear the old channels and assign the new ones, which have been used for the upgrade.
```

> Now, all the clients connected with the previous SUSE Manager Proxy can access the SUSE Manager Server again and the proxy itself can be managed by the SUSE Manager server as before.

Reviewer: ```suggestion
All the clients connected with the previous {productname} Proxy can access the {productname} Server again and the proxy itself can be managed by the server as before.
```
